src/lib/telemetry.rs

```diff
@@ -135,6 +135,31 @@ pub enum LogKey {
 }
 
 impl LogKey {
+    /// Use a string-based suffix to produce a related enum value.
+    ///
+    /// This is useful, for example, in situations where an abstract function is
+    /// used in different contexts, but needs to log to different keys depending
+    /// on the context. In this situation the abstract function can accept a
+    /// "base key" as an argument, and the function can produce new Enum values
+    /// by adding a known suffix to the base value.
+    ///
+    /// Note that the value produced by the method must be a valid Enum value.
+    /// If the stringified value is invalid, the original enum value will be
+    /// returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// pub fn abstract_logger(base_key: &LogKey) {
```

Comment:
There's no need to make this a generic example of an "abstract logger".

Just keep the 159 and 160 with what add_suffix actually returns including in the case of an unknown Enum.

Reply:
done.